api/pkg/db/validation/validation_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The test validates the `Not Null` constraints on the tables and if there's
+// any insertion of data, then it is restricted.
+package validation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"github.com/tektoncd/hub/api/pkg/testutils"
+)
+
+func TestCheckNotNull(t *testing.T) {
+	tc := testutils.Setup(t)
+
+	err := tc.Error()
+
+	db := tc.DB()
+
+	err = db.Create(&model.Catalog{Name: "tekton", Type: "", URL: "", Revision: "master"}).Error
```

Comment:
I think this will return an error only because `Type` is null and we won't know whether `URL` is `NOT NULL` or not.